x-pack/solutions/observability/plugins/synthetics/server/tasks/sync_private_locations_monitors_task.ts

```diff
@@ -415,9 +417,15 @@ export class SyncPrivateLocationMonitorsTask {
         'Skipping cleanup of duplicated package policies as it has already been done once'
       );
       return { performSync };
-    } else {
-      this.debugLog('Starting cleanup of duplicated package policies');
+    } else if (taskState.maxCleanUpRetries <= 0) {
+      this.debugLog(
+        'Skipping cleanup of duplicated package policies as max retries have been reached'
+      );
+      taskState.hasAlreadyDoneCleanup = true;
```

Comment:
Why not make this an object like you asked me to in the last PR 😛 . Truthfully, I don't really care all that much.

Reply:
Keep it like this. It was hard for me to reason about in the logic when I had an object.